erts/emulator/beam/jit/arm/beam_asm.hpp

```diff
@@ -1313,6 +1313,21 @@ class BeamModuleAssembler : public BeamAssembler {
         }
     }
 
+    bool is_bsl_small(const ArgSource &LHS, const ArgSource &RHS) {
+        if (!(always_small(LHS) && always_small(RHS))) {
+            return false;
+        } else {
+            auto [min1, max1] = getIntRange(LHS);
+            auto [min2, max2] = getIntRange(RHS);
+
+            if (min1 < 0 || max1 == 0 || min2 < 0 || min2 != max2) {
```

Comment:
`min2 != max2` looks a bit odd, won't this reject non-constant `RHS`? Is there any reason it can't work for, say `min2 = 3, max2 = 4`?